### PR TITLE
[RAPTOR-11322] Add lazy loading processor to download the associated lazy loading remote files to the local volume

### PR DESCRIPTION
**Pull request by zohar-mizrahi**
## Summary
In the context of the lazy loading custom model support, it is required to process the related environment variable, if exists, and download the required files to the root folder of the executed custom model.

## Changes
* Add a new method to the `BaseLanguagePredictor` to handle lazy loading files and call it from the `PythonPredictor` within the configuration phase.
* Add a new handler `LazyLoadingHandler` to deal with processing the lazy loading structured data JSON from an environment variable and use `datarobot_storage` to download the lazy loading files in parallel.
* Small improvement to the `credential_type` attribute definition in `S3Credentials`.
* Minor change to `.gitconfig`.
* Unit, integration and functional tests - note that the functional test requires special environment variables in order not to be skipped. It assumes an accessible MinIO service that is set with specific files. (I run it locally in my environment).

### CODE REVIEW

**Review by rvorobii — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -86,7 +86,7 @@ celerybeat-schedule
 *.sage.py
 
 # Environments
-.env
+.env*
```
Comment (akshoop) [inline .gitignore:89]: Why this change?

File: custom_model_runner/datarobot_drum/drum/lazy_loading/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class InvalidValue(Exception):
```
Comment (akshoop) [inline custom_model_runner/datarobot_drum/drum/lazy_loading/exceptions.py:1]: This seems unused

File: custom_model_runner/requirements.txt [hunk 3]
```diff
@@ -26,3 +26,4 @@ termcolor
 packaging
 markupsafe<=2.1.3
 pydantic==2.9.2
+datarobot-storage
```
Comment (akshoop) [inline custom_model_runner/requirements.txt:29]: I think this is only available internally, are we sure we can add this dependency?

File: tests/functional/test_lazy_loading.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    def skip_minio_test_if_missing_env_vars(self):
+        if "AWS_ACCESS_KEY_ID" not in os.environ:
+            pytest.skip("AWS_ACCESS_KEY_ID not set in environment")
+        if "AWS_SECRET_ACCESS_KEY" not in os.environ:
+            pytest.skip("AWS_SECRET_ACCESS_KEY not set in environment")
+        if "MINIO_ENDPOINT_URL" not in os.environ:
+            pytest.skip("Minio endpoint URL not set in q")
+
+    @pytest.fixture
+    def repository(self, skip_minio_test_if_missing_env_vars):
```
Comment (akshoop) [inline tests/functional/test_lazy_loading.py:31]: Do you need the `skip_minio_test_if_missing_env_vars` fixture here?

File: custom_model_runner/requirements.txt (hunk 3 above)
Comment (akshoop) [reply custom_model_runner/requirements.txt:29]: Would be good to pin to specific version too

File: .gitignore (hunk 1 above)
Comment (zohar-mizrahi) [reply .gitignore:89]: Because I have different env files for different environments, that is: .env.s3 and .env.minio

File: custom_model_runner/requirements.txt (hunk 3 above)
Comment (zohar-mizrahi) [reply custom_model_runner/requirements.txt:29]: Yes, we can use it - there was a decision to make it public.

Comment (zohar-mizrahi) [reply custom_model_runner/requirements.txt:29]: Ok, I'll allow any versions up to a major upgrade.

File: tests/functional/test_lazy_loading.py (hunk 4 above)
Comment (zohar-mizrahi) [reply tests/functional/test_lazy_loading.py:31]: Yes, because the given fixture tries to read expected environment variable: `os.environ[MINIO_ENDPOINT_URL]`

Comment (akshoop) [reply tests/functional/test_lazy_loading.py:31]: But it looks like there's only 1 test in this module `test_download_from_minio_success` which already uses `skip_minio_test_if_missing_env_vars` fixture. So I'm saying that having it in `repository` fixture seems redundant

File: custom_model_runner/requirements.txt [hunk 5]
```diff
@@ -26,3 +26,4 @@ termcolor
 packaging
 markupsafe<=2.1.3
 pydantic==2.9.2
+datarobot-storage>=0.0.0,<1.0.0
```
[the remark this answers is not in the document]
Comment (YuriyGuts) [reply custom_model_runner/requirements.txt:29]: Curious about the reason to pin this to `0.*` versions specifically. Does it have something to do with artifactory releases vs. OSS releases? The reason I'm asking is Buzok is at `datarobot-storage==2.0.3` and we can't bump DRUM to the latest version because of this conflict.

[the remark this answers is not in the document]
Comment (YuriyGuts) [reply custom_model_runner/requirements.txt:29]: Sorry, I should probably rather ask in #drum in case we need the input of related teams on this.
